tap_zuora/client.py

```diff
@@ -112,10 +111,14 @@ def from_config(config: Dict):
         european = config.get("european", False) == "true"
         partner_id = config.get("partner_id", None)
         is_rest = config.get("api_type") == "REST"
+        auth_type = config.get("auth_type", "Basic")
+        if auth_type in ["", None]:
+            raise BadCredentialsException("auth_type must not be set to an empty string or None")
```

Comment:
I think here we should check for valid values and raise exception for invalid values.

Reply:
```suggestion
        auth_type = config.get("auth_type", "Basic")
        if auth_type not in ["Basic", "OAuth"]:
            raise BadCredentialsException("...")
```